src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1235,6 +1210,44 @@ std::string Visitor::visitTypesafe(Parser::RdfLiteralContext* ctx) {
   return ret;
 }
 
+// ____________________________________________________________________________________
+std::variant<int64_t, double> Visitor::visitTypesafe(
+    Parser::NumericLiteralContext* ctx) {
+  return visitAlternative<std::variant<int64_t, double>>(
+      ctx->numericLiteralUnsigned(), ctx->numericLiteralPositive(),
+      ctx->numericLiteralNegative());
+}
+
+// ____________________________________________________________________________________
+namespace {
+template <typename Ctx>
+std::variant<int64_t, double> parseNumericLiteral(Ctx* ctx, bool parseAsInt) {
+  if (parseAsInt) {
+    return std::stoll(ctx->getText());
+  } else {
+    return std::stod(ctx->getText());
+  }
+}
```

Comment:
Both of these functions can throw exceptions that should be caught.
I think `invalid_argument` should not happen because of the grammar rules,
but `std::out_of_range` can occur, as arbitrarily long numbers and exponents are allowed by the SPARQL grammar. Those should be converted to readable ParseExceptions, and this case should be tested (see my comment in the test).